apps/dashboard/app/models/allowlist_policy.rb

```diff
@@ -71,19 +72,19 @@ def real_expanded_path_not_exist(path)
     Pathname.new(path).expand_path
   end
 
-  # Determine if child is a subpath of parent
-  #
-  # If the relative path from the child to the supposed parent includes '..'
-  # then child is not a subpath of parent
+  # Determine if child is a subpath of parent.
+  # This does not access the filesystem, so symlinks should be evaluated before this.
   #
   # @param parent [Pathname]
   # @param child [Pathname]
   # @return Boolean
   def child?(parent, child)
-    !child.expand_path.relative_path_from(
-      parent.expand_path
-    ).each_filename.to_a.include?(
-      '..'
-    )
+    # Expand both paths to evaluate any potential ".." components to be able to compare them as strings.
+    p = parent.expand_path.to_s
+    c = child.expand_path.to_s
+    # Child path shorter than parent => not a child.
+    return false if c.size < p.size
+    # Child path is same as parent path, or has additional components after parent path (has "/" after parent path).
+    c[0...p.size] == p && (c.size == p.size || c[p.size] == "/")
```

Comment:
This has also bee no my mind lately. Isn't this whole expression `c.start_with?(parent)`?  Looks like `start_with` is a c function so it may be even faster.

Reply:
That is true indeed. Didn't seem to be faster, but it is simpler for sure.